content/doc/book/pipeline/syntax.adoc

```diff
@@ -0,0 +1,851 @@
+    post { /* <1> */
+        always { /* <2> */
+            echo 'I will always say Hello again!'
+        }
+    }
+}
+----
+<1> Conventionally, the `post` section should be placed at the end of the
+Pipeline.
+<2> The <<post-conditions>> blocks can use steps.
+
+
+==== stages
+
+A sequence of one or more [[stage]] directives, the `stages` section is where
+the bulk of the "worK" described by a Pipeline will be located. At a minimum it
+is recommended that `stages` contain at least one [[stage]] directive for
+Build, Test, and Deploy.
+
+[cols="^10h,>90a",role=syntax]
+|===
+| Required
+| Yes
+
+| Parameters
+| _None_
+
+| Allowed
+| Only once, inside the `pipeline` block.
+|===
+
+[[stages-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    stages { /* <1> */
+        stage('Example') {
+            steps {
+                echo 'Hello World'
+            }
+        }
+    }
+}
+----
+<1> The `stages` section will typically follow the directives such as `agent`,
+`options`, etc.
+
+==== steps
+
+Defines a series of steps to be executed in a given `stage` directive.
+
+[cols="^10h,>90a",role=syntax]
+|===
+| Required
+| Yes
+
+| Parameters
+| _None_
+
+| Allowed
+| Inside each `stage` block.
+|===
+
+[[steps-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent any
+    stages {
+        stage('Example') {
+            steps { /* <1> */
+                echo 'Hello World'
+            }
+        }
+    }
+}
+----
+<1> The `steps` section must contain one or more steps.
+
+[[declarative-directives]]
+=== Directives
+
+==== agent
+
+The `agent` directive specifies where the entire Pipeline, or a specific stage,
+will execute in the Jenkins environment depending on where the `agent`
+directive is placed. The directive must be defined at the top-level inside the
+`pipeline` block, but stage-level usage is optional.
+
+
+[cols="^10h,>90a",role=syntax]
+|===
+| Required
+| Yes
+
+| Parameters
+| <<agent-parameters, Described below>>
+
+| Allowed
+| In the top-level `pipeline` block and each `stage` block.
+|===
+
+
+[[agent-parameters]]
+===== Parameters
+
+In order to support the wide variety of use-cases Pipeline authors may have,
+the `agent` directive supports a few different types of parameters. These
+parameters can be applied at the top-level of the `pipeline` block, or within
+each `stage` directive.
+
+any:: Execute the Pipeline, or stage, on any available agent. For example: `agent any`
+none:: When applied at the top-level of the `pipeline` block, each `stage`
+directive will need to contain its own `agent` directive. For example: `agent none`
+label:: Execute the Pipeline, or stage, on the agent available in the Jenkins
+environment with the provided label. For example: `agent { label 'my-defined-label' }`
+docker:: Execute the Pipeline, or stage, with the given container which will be
+dynamically provisioned on a <<../glossary#node, node>> pre-configured to
```

Comment:
There is a global and folder-based configuration option to set a Docker-capable label. I'm punting on this for right now. I think the right place to put this screenshot and information would be under Managed Jenkins and then cross-reference from here, but without that section available it's (IMO) crap to stick it into the syntax document.


As for the `node` inconsistency, I don't like it, but I think it's important to leave it. The software uses inconsistent terms :frowning: